Examples/Detectors/GeoModelDetector/src/GeoModelMuonMockupBuilder.cpp

```diff
@@ -39,211 +39,410 @@ std::unique_ptr<const Acts::TrackingGeometry>
+      static_cast<std::size_t>(FirstContainerIdx::nFirstContainers), nullptr);
+
+  // Second level: one container for the barrel and one for the two NSWs. These
+  // containers are attached to the body container and will be stacked in r.
+  std::vector<CylinderContainerBlueprintNode*> SecondContainers(
+      static_cast<std::size_t>(SecondContainerIdx::nSecondContainers), nullptr);
+
+  // Helper lambda to retrieve the first-level container
+  auto retrieveFirstContainer = [this, &cyl, &FirstContainers,
+                                 &configureContainer](
+                                    FirstContainerIdx containerIdx) {
+    auto& container = FirstContainers[static_cast<std::size_t>(containerIdx)];
+    if (container == nullptr) {
+      container =
+          &cyl.addCylinderContainer(firstContainerIdxToString(containerIdx),
+                                    (containerIdx == FirstContainerIdx::Body)
+                                        ? Acts::AxisDirection::AxisR
+                                        : Acts::AxisDirection::AxisZ);
+      configureContainer(*container);
+    }
+    return container;
+  };
+
+  // Helper lambda to retrieve the second-level container
+  auto retrieveSecondContainer = [&FirstContainers, &SecondContainers,
+                                  &configureContainer](bool isBarrel) {
+    auto& container = SecondContainers[static_cast<std::size_t>(
+        isBarrel ? SecondContainerIdx::Barrel : SecondContainerIdx::NSWs)];
+    if (container == nullptr) {
+      auto& bodyContainer =
+          *FirstContainers[static_cast<std::size_t>(FirstContainerIdx::Body)];
+      container = &bodyContainer.addCylinderContainer(
+          isBarrel ? "BarrelContainer" : "NSWsContainer",
+          isBarrel ? Acts::AxisDirection::AxisR : Acts::AxisDirection::AxisZ);
+      configureContainer(*container);
+    }
+    return container;
+  };
+
+  // Sorting the boxes by station
+  std::sort(boundingBoxes.begin(), boundingBoxes.end(),
+            [this](const auto& a, const auto& b) {
+              return getStationIdx(a) < getStationIdx(b);
+            });
 
-  // Add the station nodes as static cylidner nodes
+  auto it = boundingBoxes.begin();
   std::size_t layerId = 1;
+  while (it != boundingBoxes.end()) {
+    // Current station index
+    StationIdx currentIdx = getStationIdx(*it);
+    bool isBarrel = (static_cast<int>(currentIdx) < 3);
+
+    // Find the range of boxes for the current station
+    auto rangeEnd = std::find_if(it, boundingBoxes.end(),
+                                 [this, currentIdx](const auto& box) {
+                                   return getStationIdx(box) != currentIdx;
+                                 });
+
+    // Check we want to build this station
+    const std::string station = stationIdxToString(currentIdx);
+    if (std::find(m_cfg.stationNames.begin(), m_cfg.stationNames.end(),
+                  station) == m_cfg.stationNames.end()) {
```

Comment:
```suggestion
    if (!rangeContainsValue(m_cfg.stationNames, station)) {
```
Ensure that https://github.com/acts-project/acts/blob/main/Core/include/Acts/Utilities/Helpers.hpp is included